modelskill/timeseries/_timeseries.py

```diff
@@ -271,15 +291,28 @@ def trim(
         start_time = pd.Timestamp(start_time) - pd.Timedelta(buffer)
         end_time = pd.Timestamp(end_time) + pd.Timedelta(buffer)
 
-        self.data = self.data.sel(time=slice(start_time, end_time))
+        # TODO: return a new TimeSeries object instead of modifying the existing one
+
+        data = self.data.sel(time=slice(start_time, end_time))
+        if len(data.time) == 0:
+            raise ValueError(
+                f"No data left after trimming to {start_time} - {end_time}"
+            )
+        self.data = data
 
-    def interp_time(self, new_time: pd.DatetimeIndex) -> TimeSeries:
+    def interp_time(
```

Comment:
There seems to be a flaw in our design. 😳

* It makes for a `TimeSeries` to be able to interpolate.
* It doesn't make sense to interpolate in track data, e.g. `TrackModelResult` and `TrackObservation`
* Thus, track data are not `TimeSeries` (at least in this sense) 🤔

Reply:
Hmm, maybe we should have dedicated TrackTimeseries and PointTimeseries after all?